src/token/fa2.mligo

```diff
@@ -80,14 +80,7 @@ let credit_to (amt, to_, token_id, ledger, total_supply : nat * address * nat *
 
 let transfer_item (store, ti : storage * transfer_item): storage =
   let transfer_one (store, tx : storage * transfer_destination): storage =
-    let valid_from_ =
-      if tx.token_id = store.frozen_token_id
-      then
-        if Tezos.sender = store.admin
-        then ti.from_
-        else (failwith("FROZEN_TOKEN_NOT_TRANSFERABLE") : address)
-      else check_sender (ti.from_, tx.token_id, store)
-      in
+    let valid_from_ = check_sender (ti.from_, tx.token_id, store) in
```

Comment:
As said above we should still keep the check, however since it is now simpler I suggest moving the `if tx.token_id = store.frozen_token_id then (failwith("FROZEN_TOKEN_NOT_TRANSFERABLE") : address)` inside `check_sender` itself.

Reply:
I've updated this as well.